lib/jekyll/entry_filter.rb

```diff
@@ -7,6 +7,8 @@ class EntryFilter
       ".", "_", "#", "~",
     ].freeze
 
+    SPECIAL_LEADING_CHAR_REGEX = %r!\A#{Regexp.union(SPECIAL_LEADING_CHARACTERS)}!o.freeze
```

Comment:
This is the only place we use `SPECIAL_LEADING_CHARACTERS` in this class, so we can just remove it.